src/SonarScanner.MSBuild.Tasks/IsTestByReference.cs

```diff
@@ -84,7 +84,7 @@ private string ParseName(string fullName)
         {
             try
             {
-                return new AssemblyName(fullName).Name;
+                return new AssemblyName(fullName.Trim('@', '(', ')')).Name;
```

Comment:
Does it actually execute the trim for every char in the order ? Otherwise maybe the first parenthesis should be treated after this first @ and ) round.

Reply:
This `Trim` doesn't care about the order. It trims **all** such characters from begin and end. It would trim all of these to `x`:
```
@x@
))x((
(@)(@)x)@@(
@@(())x@@(())
..and so on
```

Reply:
Ok got my answer while reading the method doc, so it's ok :)